parser/src/ast/pub_inputs.rs

```diff
@@ -27,4 +27,8 @@ impl PublicInput {
     pub fn size(&self) -> usize {
         self.size
     }
+
+    pub fn into_parts(self) -> (String, usize) {
```

Comment:
optional minor nit: Might be good to add docs or a todo here and other places.

Reply:
Yeah, we should have docs, but the whole codebase will need a walk-through for docs updates, which I'm planning to do this afterwards when it won't be blocking anyone else. I didn't want to spend too much time checking these things step-by-step when I was reorganizing the commits and splitting this for the multiple refactors since it'll need a solid review anyway.